ganttproject/src/biz/ganttproject/impex/csv/XlsReaderImpl.java

```diff
@@ -53,11 +60,50 @@ public void close() throws IOException {
 
   @Override
   public Iterator<SpreadsheetRecord> iterator() {
-    return Iterators.transform(myBook.getSheetAt(0).iterator(), (input) -> new XlsRecordImpl(getCellValues(input), myHeaders));
+    Iterator<SpreadsheetRecord> iterator = null;
+
+    if(myBook.getNumberOfSheets() > 1) {
+      Iterator<SpreadsheetRecord> it = null;
+      for (int i = 0; i < myBook.getNumberOfSheets(); ++i) {
+        Sheet sheet = myBook.getSheetAt(i);
+        if (i < myBook.getNumberOfSheets() - 1) {
+          int lastRow = sheet.getPhysicalNumberOfRows();
+          sheet.createRow(lastRow + 1);
+          sheet.createRow(lastRow + 2);
+        }
+        it = Iterators.transform(sheet.iterator(), (input) -> new XlsRecordImpl(getCellValues(input), myHeaders));
+        iterator = (iterator == null) ? it : Iterators.concat(iterator, it);
+      }
+    }
+    else {
+      iterator = Iterators.transform(myBook.getSheetAt(0).iterator(), (input) -> new XlsRecordImpl(getCellValues(input), myHeaders));
+    }
+
+    return iterator;
   }
 
   private List<String> getCellValues(Row row) {
-    return Lists.newArrayList(Iterables.transform(row, Cell::getStringCellValue));
+    return Lists.newArrayList(Iterables.transform(row,
+            (input) -> {
+              switch (input.getCellTypeEnum()) {
+                case NUMERIC: {
+                  if (HSSFDateUtil.isCellDateFormatted(input)) {
+                    return myDateFormat.format(input.getDateCellValue());
+                  }
+                  else {
+                    if (input.getCellStyle().getDataFormat() == (short)1) {
```

Comment:
This looks as if it produces the same result as subsequent else, no? Can we just call `String.valueOf(input.getNumericCellValue())` no matter what data format is? Or even just call `getStringCellValue` in all cases when `isCellDateFormatted` returns false?